source/extensions/common/wasm/wasm.cc

```diff
@@ -2514,6 +2600,38 @@ void Wasm::queueReady(uint32_t root_context_id, uint32_t token) {
   it->second->onQueueReady(token);
 }
 
+Network::FilterStatus Context::onNewConnection() { return onNetworkNewConnection(); };
+
+Network::FilterStatus Context::onData(Buffer::Instance& data, bool end_stream) {
+  network_downstream_data_buffer_ = &data;
+  auto result = onDownstreamData(data.length(), end_stream);
+  network_downstream_data_buffer_ = nullptr;
+  return result;
+}
+
+Network::FilterStatus Context::onWrite(Buffer::Instance& data, bool end_stream) {
+  network_upstream_data_buffer_ = &data;
+  auto result = onUpstreamData(data.length(), end_stream);
+  network_upstream_data_buffer_ = nullptr;
+  return result;
+}
+
+void Context::onEvent(Network::ConnectionEvent event) {
+  if (event == Network::ConnectionEvent::LocalClose ||
+      event == Network::ConnectionEvent::RemoteClose) {
+    onConnectionClosed();
```

Comment:
Shouldn't we differentiate these two?  You could pass in a direction field (for example).

Reply:
Done.